quipucords/api/aggregate_report/view.py

```diff
@@ -1,71 +1,32 @@
-    jboss_ws_cores_virtual = serializers.FloatField(default=0.0, min_value=0)
-    jboss_ws_instances = serializers.IntegerField(default=0, min_value=0)
-    openshift_cores = serializers.IntegerField(default=0, min_value=0)
-    openshift_operators_by_name = serializers.JSONField(default=dict)
-    openshift_operators_by_kind = serializers.JSONField(default=dict)
-    os_by_name_and_version = serializers.JSONField(default=dict)
-    socket_pairs = serializers.IntegerField(default=0, min_value=0)
-    system_creation_date_average = serializers.DateField()
-    vmware_hosts = serializers.IntegerField(default=0, min_value=0)
-    vmware_vm_to_host_ratio = serializers.FloatField(default=0)
-    vmware_vms = serializers.IntegerField(default=0, min_value=0)
-    openshift_cluster_instances = serializers.IntegerField(default=0, min_value=0)
-    openshift_node_instances = serializers.IntegerField(default=0, min_value=0)
-
-
-class DiagnosticsSerializer(serializers.Serializer):
-    """Serializer for the diagnostics portion of the Aggregate Report."""
-
-    inspect_result_status_failed = serializers.IntegerField(default=0, min_value=0)
-    inspect_result_status_success = serializers.IntegerField(default=0, min_value=0)
-    inspect_result_status_unknown = serializers.IntegerField(default=0, min_value=0)
-    inspect_result_status_unreachable = serializers.IntegerField(default=0, min_value=0)
-    missing_cpu_core_count = serializers.IntegerField(default=0, min_value=0)
-    missing_cpu_socket_count = serializers.IntegerField(default=0, min_value=0)
-    missing_name = serializers.IntegerField(default=0, min_value=0)
-    missing_pem_files = serializers.IntegerField(default=0, min_value=0)
-    missing_system_creation_date = serializers.IntegerField(default=0, min_value=0)
-    missing_system_purpose = serializers.IntegerField(default=0, min_value=0)
-
-
-class AggregateReportSerializer(serializers.Serializer):
-    """Aggregate Report Serializer."""
-
-    results = ResultsSerializer(many=False)
-    diagnostics = DiagnosticsSerializer(many=False)
+from api.aggregate_report.serializer import AggregateReportSerializer
 
 
 @extend_schema(responses=AggregateReportSerializer)
 @api_view(["GET"])
 def aggregate_report(request, report_id: int) -> Response:
-    """Lookup and return a details system report."""
-    report_dict = get_aggregate_report_by_report_id(report_id)
-    if report_dict:
-        return Response(report_dict)
+    """Lookup and return an Aggregate Report."""
+    aggregate = get_serialized_aggregate_report(report_id)
+    if aggregate:
+        return Response(aggregate)
     raise NotFound
+
+
+def get_serialized_aggregate_report(report_id: int) -> AggregateReportSerializer | None:
+    """
+    Get the aggregate report for the report id specified.
+
+    If the aggregate report does not exist for the report specified,
+    it will be created.
+    """
+    aggregate = get_aggregate_report_by_report_id(report_id)
```

Comment:
don't need to change anything if you don't want right now, but for v2 API I definitely would like to see the possibility of a automatic generation at request time to be completely gone UNLESS that has a completely async workflow. But honestly, I would just throw a error at it and have the user manually regenerating the whole report.

With the changes introduced in this PR, aggregate report generation should be part of the "fingerprinting" process and always be present in the report. If for some reason it wasn't generated, it should be considered an error IMO.

Reply:
Per my other comment, after rereading the JIRA story for this issue I noticed it specifically calls out for simply throwing an error if the report does not exist.

Reply:
Per our discussion on Red Hat's slack, we will change this behavior only in v2 api.